lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -0,0 +1,132 @@
+import path from "node:path";
+import os from "node:os";
+import AbstractResolver from "./AbstractResolver.js";
+import Installer from "./maven/Installer.js";
+import {getLogger} from "@ui5/logger";
+const log = getLogger("ui5Framework:Sapui5MavenSnapshotResolver");
+
+const DIST_PKG_NAME = "@sapui5/distribution-metadata";
+const DIST_GROUP_ID = "com.sap.ui5.dist";
+const DIST_ARTIFACT_ID = "sapui5-sdk-dist";
+
+/**
+ * Resolver for the SAPUI5 framework
+ *
+ * @public
+ * @class
+ * @alias @ui5/project/ui5Framework/Sapui5MavenSnapshotResolver
+ * @extends @ui5/project/ui5Framework/AbstractResolver
+ */
+class Sapui5MavenSnapshotResolver extends AbstractResolver {
+	/**
+	 * @param {*} options options
+	 * @param {string} options.snapshotEndpointUrl Maven Repository Snapshot URL
+	 * @param {string} options.version SAPUI5 version to use
+	 * @param {string} [options.sources=false] Whether to install framework libraries as sources or
+	 * pre-built (with build manifest)
+	 * @param {string} [options.cwd=process.cwd()] Working directory to resolve configurations like .npmrc
```

Comment:
This is copied from AbstractResolver, but as npm is not used, the example of `.npmrc` might be confusing.

Reply:
Furthermore, it's passed to the mvn/Installer and then to mvn/Registry. Within the Registry, there's also a check for it and if missing, an Error is thrown. However, in this scenario, it doesn't seem to be needed at all.

Do you think we might remove it entirely?

Reply:
Removed the `cwd` property for this scenario, because it was completely redundant